bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+import Keyring from '@polkadot/keyring';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  getChainflipApi,
+  observeEvent,
+  lpMutex,
+  chainFromAsset,
+  shortChainFromAsset,
+  amountToFineAmount,
+  assetDecimals,
+  Event,
+  ingressEgressPalletForChain,
+  submitChainflipExtrinsic,
+  calculateFeeWithBps,
+  amountToFineAmountBigInt,
+  newAddress,
+} from './utils';
+import { send } from './send';
+import { provideLiquidity } from './provide_liquidity';
+import { requestNewSwap } from './perform_swap';
+import { createBoostPools } from './setup_boost_pools';
+
+const keyring = new Keyring({ type: 'sr25519' });
+keyring.setSS58Format(2112);
+
+async function lpSubmitExtrinsic(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  extrinsic: any,
+  keyringPair: KeyringPair,
+  errorOnFail: boolean = true,
+) {
+  let extrinsicResult;
+  await lpMutex.runExclusive(async () => {
+    extrinsicResult = await submitChainflipExtrinsic(keyringPair, extrinsic, errorOnFail);
+  });
+  return extrinsicResult;
+}
+
+/// Stops boosting for the given boost pool tier and returns the StoppedBoosting event.
+export async function stopBoosting(
+  asset: Asset,
+  boostTier: number,
+  lpUri = '//LP_BOOST',
+  errorOnFail: boolean = true,
+): Promise<Event | undefined> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  const observeStoppedBoosting = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:StoppedBoosting',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  const extrinsicResult = await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].stopBoosting(
+      shortChainFromAsset(asset).toUpperCase(),
+      boostTier,
+    ),
+    lp,
+    errorOnFail,
+  );
+  if (!extrinsicResult.dispatchError) {
+    console.log('waiting for stop boosting event');
+    return observeStoppedBoosting;
+  }
+  console.log('Already stopped boosting');
+  return undefined;
+}
+
+/// Provides liquidity, then adds the funds to the boost pool of the given tier and returns the BoostFundsAdded event.
+export async function addBoostFunds(
+  asset: Asset,
+  boostTier: number,
+  amount: number,
+  lpUri = '//LP_BOOST',
+): Promise<Event> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  await provideLiquidity(asset, amount, true, lpUri);
```

Comment:
I know this functions already existed, but this just adds funds to the LP account, right? I think we use "provide liquidity" to mean creating a range/limit order, which isn't done here. Maybe "deposit_liquidity" (or just "add_lp_funds") would be a better name?

Reply:
I didn't know that was an implied meaning. can someone else confirm? @kylezs .
Also might leave it to another PR, this one is big enough.

Reply:
I think it's ok as is, we have terms like "setRangeOrder" "setLimitOrder" for the other things. Maybe `depositLiquidity` is a little more explicit though. Think it can be left for another PR though, will leave that decision to you @j4m1ef0rd

Reply:
Happy to rename it in a different PR.

Reply:
https://linear.app/chainflip/issue/PRO-1382/rename-provideliquidity-in-bouncer-to-depositliquidity